src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -90,22 +90,26 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
                 var typeInferenceService = document.GetLanguageService<ITypeInferenceService>();
                 Contract.ThrowIfNull(typeInferenceService, nameof(typeInferenceService));
 
-                var types = typeInferenceService.InferTypes(semanticModel, position, cancellationToken);
+                var infos = typeInferenceService.GetTypeInferenceInfo(semanticModel, position, cancellationToken);
 
-                if (types.Length == 0)
-                    types = ImmutableArray.Create<ITypeSymbol>(semanticModel.Compilation.ObjectType);
+                if (infos.Length == 0)
+                    infos = ImmutableArray.Create(new TypeInferenceInfo(semanticModel.Compilation.ObjectType));
 
-                foreach (var type in types)
-                    await HandleSingleTypeAsync(context, semanticModel, token, type, cancellationToken).ConfigureAwait(false);
+                foreach (var (type, isParams) in infos)
+                    await HandleSingleTypeAsync(context, semanticModel, token, type, isParams, cancellationToken).ConfigureAwait(false);
             }
             catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, ErrorSeverity.General))
             {
                 throw ExceptionUtilities.Unreachable();
             }
         }
 
-        private static async Task HandleSingleTypeAsync(CompletionContext context, SemanticModel semanticModel, SyntaxToken token, ITypeSymbol type, CancellationToken cancellationToken)
+        private static async Task HandleSingleTypeAsync(
+            CompletionContext context, SemanticModel semanticModel, SyntaxToken token, ITypeSymbol type, bool isParams, CancellationToken cancellationToken)
         {
+            if (isParams && type is IArrayTypeSymbol arrayType)
```

Comment:
Doesn't this fix more than just enum types? If so, can you add a ref type test too?